library/core/src/bool.rs

```diff
@@ -60,4 +60,50 @@ impl bool {
     pub fn then<T, F: FnOnce() -> T>(self, f: F) -> Option<T> {
         if self { Some(f()) } else { None }
     }
+
+    /// Returns either `true_val` or `false_val` depending on the value of
+    /// `condition`, with a hint to the compiler that `condition` is unlikely
+    /// to be correctly predicted by a CPU’s branch predictor.
+    ///
+    /// This method is functionally equivalent to writing
+    /// ```ignore (this is just for illustrative purposes)
+    /// if b { true_val } else { false_val }
+    /// ```
```

Comment:
This is technically correct in that *the function's body* could be replaced by this (with s/b/self/) but it's subtly misleading when looking at call sites, which will be the POV of most readers. Both `true_val` or `false_val` are always evaluated unconditionally, and this is semantically important for side effects and [enables the branch-free lowering in the first place when side effects may exist](https://github.com/rust-lang/libs-team/issues/468#issuecomment-2431402691).

Technically that's not specific to this function, you always have to keep it in mind when manually inlining a function (and e.g. rust-analyzer does just that). But it's especially important here: both arguments being evaluated unconditionally is part of the point of having this function. So the docs should call attention to it, either giving the fully correct inlined version (evaluating all three arguments before the `if`) or by side-stepping the pitfalls source-level inlining, e.g.,:
```suggestion
    /// This method is functionally equivalent to
    /// ```ignore (this is just for illustrative purposes)
    /// fn select_unpredictable<T>(b: bool, true_val: T, false_val: T) -> T {
    ///     if b { true_val } else { false_val }
    /// }
    /// ```
```